core/corehttp/ipns_hostname.go

```diff
@@ -19,8 +19,8 @@ func IPNSHostnameOption() ServeOption {
 			defer cancel()
 
 			host := strings.SplitN(r.Host, ":", 2)[0]
-			if p, err := n.Namesys.Resolve(ctx, host); err == nil {
-				r.URL.Path = "/ipfs/" + p.String() + r.URL.Path
+			if p, err := n.Namesys.Resolve(ctx, host, 0); err == nil {
+				r.URL.Path = p.String() + r.URL.Path
```

Comment:
Other solution which should also fix #1234 would be to change this to `/ipns/` path and let the IPNS stuff do the magic. (For IPNS there is already an exception for `Cache-Control`)

```diff
diff --git a/core/corehttp/ipns_hostname.go b/core/corehttp/ipns_hostname.go
index 7361001..0da667c 100644
--- a/core/corehttp/ipns_hostname.go
+++ b/core/corehttp/ipns_hostname.go
@@ -19,8 +19,8 @@ func IPNSHostnameOption() ServeOption {
                        defer cancel()

                        host := strings.SplitN(r.Host, ":", 2)[0]
-                       if p, err := n.Namesys.Resolve(ctx, host); err == nil {
-                               r.URL.Path = "/ipfs/" + p.String() + r.URL.Path
+                       if _, err := n.Namesys.Resolve(ctx, host); err == nil {
+                               r.URL.Path = "/ipns/" + host + r.URL.Path
                        }
                        childMux.ServeHTTP(w, r)
                })
```